### PR TITLE
Add descriptive class names to DAQ components.

### LINKED ISSUE #36: Precision Input text not visible during entry
Just looking at the user guide: https://dash.plot.ly/dash-daq/precisioninput, text is not visible during entry in the field. Surely, this is not intended? 

Using Safari Browser here

### LINKED ISSUE #55: daqNumericInput is not compatible with Dark Theme
`daqNumericInput ` remains white on dark mode. Inside text is converted to white. This results white input on white background hence not readable. Please see below a reproducible R example and associated screenshots.

```r
library(dash)
library(dashDaq)
library(dashHtmlComponents)

app <- Dash$new()

theme <- list(
  'dark' = TRUE,
  'detail' = '#007439',
  'primary' = '#00EA64',
  'secondary' = '#6E6E6E'
)

rootLayout <- htmlDiv(list(
  daqLEDDisplay(value = "3.14159"),
  htmlBr(),
  daqNumericInput(min = 0, max = 10, value = 4),
  htmlBr(),
  daqPowerButton(on = TRUE)
))

app$layout(htmlDiv(
  list(daqDarkThemeProvider(theme = theme,
                            children = rootLayout)),
  style = list('backgroundColor' = '#303030',
                     'float' = 'left')
))

app$run_server(port = 8897)
```

Screen Shot 2019-07-10 at 2 38 39 PM
Screen Shot 2019-07-10 at 2 38 49 PM

### LINKED ISSUE #63: Overwrite auto-generated className for nested components
Inherited from `styled-components`, all daq components up to v0.1.7 are using auto-generated className for nested components. as shown below:
daq

It will be good to have component-specific className for easier tracking/styling in DOM tree.

cc @wbrgss

### PR DESCRIPTION
**Pull request by shammamah**
Closes #63. 
Closes #36. 
Closes #55.

## About 
* Added class names to components according to BEM conventions: `daq-[component name]--[light/dark]__[component element/modifiers on those elements]`
* Stopped unnecessary props (`className`, `id`, and `setProps`) from being passed to label containers, as this would cause the containers to have the same `className`s as the components
* Fixed colour problems for dark-theme buttons and inputs (could use some feedback on the chosen colours here)
* Used `withTheme` for components that didn't use it, so that they would have the correct class names
* A bit of code cleanup

### Dark theme components 
Before: 
Screen Shot 2020-01-17 at 4 35 51 PM
(after clicking on the `PrecisionInput` to enter a value)
Screen Shot 2020-01-17 at 4 35 59 PM

After: 
Screen Shot 2020-01-17 at 4 36 41 PM
Screen Shot 2020-01-17 at 4 36 49 PM

### CODE REVIEW

File: src/components/PowerButton.react.js [hunk 1]
```diff
@@ -35,28 +37,30 @@ class PowerButton extends Component {
...
+
+    const INDICATOR_SIZE = (size / 10) * 1.25;
 
+    const elementName = getClassName('powerbutton', theme);
+    const filteredProps = getFilteredProps(this.props);
     return (
-      <div id={this.props.id} className={this.props.className} style={this.props.style}>
-        <LabelContainer {...this.props}>
-          <Container>
+      <div id={id} className={elementName + ' ' + (className || '')} style={style}>
```
Comment (wbrgss) [inline src/components/PowerButton.react.js]: The template strings might come handy the more complicated the evaluation logic gets, like on this line

File: src/components/Gauge.react.js [hunk 2]
```diff
@@ -48,24 +51,33 @@ class Gauge extends React.Component {
 
     const progress = computeProgress({ min, max, value, progressionTarget: 1 });
 
+    const elementName = getClassName('gauge', theme);
+
     const currentValue = (
       <CurrentValue
+        className={elementName + '__current-value'}
```
Comment (wbrgss) [inline src/components/Gauge.react.js]: @shammamah Style comment, consider using [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) e.g.

``className={`${elementName}__current-value`}``

(and throughout). 

I don't really mind the string concatenation though, so if that's a conscious choice it looks clean to me.

File: src/components/Gauge.react.js [hunk 3]
```diff
@@ -48,24 +51,33 @@ class Gauge extends React.Component {
...
         css={'transform: translateY(-150%); top: 0;'}
       >
         {logarithmic ? log.formatValue(value, base) : value.toFixed(1)}
       </CurrentValue>
     );
+    const filteredProps = getFilteredProps(this.props);
 
     return (
-      <div id={id} className={className} style={style}>
+      <div id={id} className={elementName + ' ' + (className || '')} style={style}>
```
Comment (alexcjohnson) [inline src/components/Gauge.react.js]:
```suggestion
      <div id={id} className={elementName + (className ? (' ' + className) : '')} style={style}>
```
(and similar in a few other places)

File: src/components/Gauge.react.js [hunk 4]
```diff
@@ -48,24 +51,33 @@ class Gauge extends React.Component {
...
+          className={elementName + '__label'}
+          {...filteredProps}
           labelCSS={this.props.labelPosition === 'top' ? null : 'transform: translateY(-80px);'}
         >
           <Container color={colorValue}>
-            <GaugeSVG {...{ ...this.props, scale }} progress={progress} />
+            <GaugeSVG
+              className={elementName + '__gauge'}
+              {...{ ...filteredProps, scale }}
+              progress={progress}
```
Comment (alexcjohnson) [inline src/components/Gauge.react.js]: the double `...` looks funny, but if we're going to use it, might as well go all the way
`{...{ ...filteredProps, scale, progress}}`

File: src/components/PrecisionInput.react.js [hunk 5]
```diff
@@ -126,25 +132,32 @@ class PrecisionInput extends Component {
...
-        id={this.props.id}
-        className={this.props.className}
-        style={Object.assign({}, defaultRootStyles, this.props.style)}
+        id={id}
+        className={elementName + ' ' + (className || '')}
+        style={Object.assign({}, defaultRootStyles, style)}
       >
-        <LabelContainer {...this.props}>
+        <LabelContainer className={elementName + '__label'} {...filteredProps}>
           {this.state.isInput ? precisionInput : precisionOutput}
```
Comment (alexcjohnson) [inline src/components/PrecisionInput.react.js]: Not new here so nonblocking, but it's a bit weird to create both elements and then only use one. Would be better to do something like
```js
let precisionItem;
if (this.state.isInput) {
   precisionItem = <PrecisionInput ... />;
}
else {
    precisionItem = <PrecisionOutput ... />;
}
```

File: src/components/PowerButton.react.js (hunk 1 above)
Comment (alexcjohnson) [reply src/components/PowerButton.react.js]: Based on https://github.com/plotly/dash-daq/pull/80#discussion_r387853372 I think template strings would be more complicated here.

**Pull request merged**